bescort.lic

```diff
@@ -765,6 +771,88 @@ class Bescort
       current_room = current_room.wander
     end
   end
+  
+  def gate_of_souls(mode)
+    Flags.add('maze-reset', 'Plumes of lava belch into the air, sending molten rock spraying about')
+    case mode
+    when /blasted/i
+      unless Room.current.id == 1784
+        echo('Must start at Gate of Souls, room number 1784')
+        exit
+      end
+      gos_blasted_plains
+      find_room_maze
+    when /temple/i
+      unless Room.current.id == 1784
+        echo('Must start at Gate of Souls, room number 1784')
+        exit
+      end
+      gos_blasted_plains
+      wander_maze_until('golden sandstone temple', 'go temple')
+      manual_go2(13625)
+      find_room_list(%w[sw e e e se s sw w w w w nw n ne se s ne se n ne w nw])
+    when /fangs/i
+      unless Room.current.id == 1784
+        echo('Must start at Gate of Souls, room number 1784')
+        exit
+      end
+      gos_blasted_plains
+      wander_maze_until('golden sandstone temple', 'go temple')
+      manual_go2(13643)
+      fput('climb cliff')
+      find_room_maze
+    when /exit/i
+      if XMLData.room_title.include?('The Fangs of Ushnish')
+        wander_maze_until('steep cliff', 'climb cliff')
+        gos_temple_leave
+      end
+      if XMLData.room_title.include?('Temple of Ushnish')
+        gos_temple_leave
+      end
+      if XMLData.room_title.include?('Blasted Plain')
+        gos_plains_leave
+      end
+      if XMLData.room_title.include?('Before the Gate of Souls')
+        gos_tunnel
+      end
+    end
+  end
+
+  def gos_blasted_plains
+      unless DRRoom.room_objs.include?('low tunnel')
+        push_boulder
+      end
+      retreat
+      gos_tunnel
+      fput('go field')
+    end
+  
+  def push_boulder
+    fix_standing
+    fput('push boulder', 'You lean against the boulder and push with all your might', 'You\'re already pushing')
+    case waitfor('a low tunnel is revealed', 'You stop pushing', 'blocking all access to a low tunnel')
+    when 'You stop pushing', 'blocking all access to a low tunnel'
+      push_boulder
+    end
+  end
+  
+  def gos_tunnel
+    fput('kneel')
+    fput('go tunnel')
```

Comment:
💡 Should this be `move(...)`?

Reply:
Move was causing a timeout message since it takes longer than 30 seconds.  While it didn't impact the functionality, I didn't want to cause concern from users.  The other movements using FPut was just carelessness.  Added a pull to address those.